pages/docs/reference/java-to-kotlin-interop.md

```diff
@@ -397,53 +397,32 @@ public class BB8 implements Robot {
-Depending on the case of adding the annotation, specify one of the argument values:
-
-* `-Xjvm-default=enabled` should be used if you add only new methods with the `@JvmDefault` annotation.
-   This includes adding the entire interface for your API.
-* `-Xjvm-default=compatibility` should be used if you are adding a `@JvmDefault` to the methods that were available in the API before.
-   This mode helps avoid compatibility breaks: all the interface implementations written for the previous versions will be fully compatible with the new version.
-   However, the compatibility mode may add some overhead to the resulting bytecode size and affect the performance.
-
-For more details about compatibility issues, see the `@JvmDefault` [reference page](/api/latest/jvm/stdlib/kotlin.jvm/-jvm-default/index.html).
-
-### Using in delegates
-
-Note that if an interface with `@JvmDefault` methods is used as a [delegate](/docs/reference/delegation.html),
-the default method implementations are called even if the actual delegate type provides its own implementations.
-
-<div class="sample" markdown="1" theme="idea" data-highlight-only>
-
-```kotlin
-interface Producer {
-    @JvmDefault fun produce() {
-        println("interface method")
-    }
-}
+>**Note**: Prior to Kotlin 1.4, to generate default methods, you could use the `@JvmDefault` annotation on these methods.
+> Compiling with `-Xjvm-default=all` in 1.4 generally works as if you annotated all non-abstract methods of interfaces
+> with `@JvmDefault`and compiled with `-Xjvm-default=enable`. However, there are cases when their behavior differs.
+> The detailed information about changes in default methods generation in Kotlin 1.4 is provided in [this post](https://blog.jetbrains.com/kotlin/2020/07/kotlin-1-4-m3-generating-default-methods-in-interfaces/)
+> on the Kotlin blog.
 
-class ProducerImpl: Producer {
-    override fun produce() {
-        println("class method")
-    }
-}
+### Compatibility mode for default methods
 
-class DelegatedProducer(val p: Producer): Producer by p {
-}
+If there are clients that use your Kotlin interfaces compiled without the new `-Xjvm-default=all` option, then they can
+be incompatible with the same code compiled with this option. 
 
-fun main() {
-    val prod = ProducerImpl()
-    DelegatedProducer(prod).produce() // prints "interface method"
-}
-```
-</div>
+To avoid breaking the compatibility with such clients, compile your Kotlin code in the _compatibility mode_ by specifying
+the `-Xjvm-default=all-compatibility` compiler option. In this case, all the code that uses the previous version will 
+ work fine with the new one. However, the compatibility mode adds some overhead to the resulting bytecode size.
 
-For more details about interface delegation in Kotlin, see [Delegation](/docs/reference/delegation.html).
+There is no need in compatibility for the new interfaces you add because no client code uses their previous versions.
```

Comment:
not sure what this means "There is no need in compatibility for the new interfaces you add because no client code uses their previous versions."

There is no need to consider compatibility for new interfaces, as no clients have used it before